### DISCUSSION TITLE
State persistent

### DISCUSSION THREAD

**Discussion by mohittalele** [ag2ai/faststream#1578] [General]
Hello, 

I am new to library. I have use case where I would be reading from kafka topic and would process it. 
The question I have is - 
1. Is there some stateful mechanism ( like rockDB ) which I can use in case non-transient failure in processing of msg ? I know faust had something on similar lines. 
2. Whats the best way within faststream to handle non-transient failures ?

**Comment by Lancetnik** [comment]
Hello! FastStream doesn't have any kind of State excluding state, already provided by the broker (Kafka reading offset)
I am not sure, that kind of failures u talking about.
The common practice in FastStream is just read messages with consumer group (Kafka remembers you reading position between restarts in this case) and commit offset only after correct message processing. Also, you can retry any message N times and just skip it without processing if it was broken.
